### ISSUE TITLE
`no-implicit-coercion` does not check template literals

### ISSUE THREAD

**Issue by kaicataldo** [eslint/eslint#7062]
**The version of ESLint you are using**
Master

**The problem you want to solve.**
While working on https://github.com/eslint/eslint/pull/7060 I noticed that `no-implicit-coercion` does not check for implicit coercion using template literals. While this isn't a common practice, I think we should add it for the sake of completeness.

**Your take on the correct solution to problem.**
`no-implicit-coercion` should warn when it finds the following patterns:
```js
`` + foo; // Error: use `String(foo)` instead.
foo + ``; //  Error: use `String(foo)` instead.
foo += ``; // Error: use `foo = String(foo)` instead.
```

I have a local branch with an implementation and tests and am happy to make a PR once https://github.com/eslint/eslint/pull/7060 has been merged.

**Comment by platinumazure** [comment]
:+1: from me.

**Comment by vitorbal** [comment]
:+1:!!

**Comment by kaicataldo** [comment]
@eslint/eslint-team Any other thoughts on this? Could use one more supporter/would love to hear any arguments against.

**Comment by mikesherov** [comment]
It's a bug if it's not checking Template Literals. Should probably also check tagged template literals too. 

:+1:

**Comment by mikesherov** [comment]
Err, I guess not for tagged template expressions because those could return non strings. Nvm.

**Comment by ilyavolodin** [comment]
:+1:

**Comment by kaicataldo** [comment]
Thanks all - I'll put together a PR
